src/ASDF_generate_provenance.c

```diff
@@ -24,6 +24,10 @@
 #include "gen_sf_parfile_provenance.h"
 #include "ASDF_generate_provenance.h"
 
+#ifdef __cplusplus
```

Comment:
This should only be necessary in the header file. If you're compiling `.c` files as C++, then you can't expect much to work anyway.